app/Http/Controllers/Admin/AmaAnnouncementsController.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use App\AmaAnnouncement;
+use View;
+use Session;
+use Redirect;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Support\Facades\Input;
+use Auth;
+
+class AmaAnnouncementsController extends Controller
+{
+        /**
+         * Display a listing of the resource.
+         *
+         * @return Response
+         */
+        public function index()
+    {
+        // get all announcements
+        $ama_announcements = AmaAnnouncement::all();
+
+        // load the view and pass the announcment
+        return View::make('ama_announcements.index')
+            ->with('ama_announcements', $ama_announcements);
+    }
+
+    
+        /**
+     * Show the form for creating a new resource.
+     *
+     * @return Response
+     */
+    public function create()
+    {
+        // load the create form (app/resources/views/ama_announments/create.blade.php)
+        return View::make('ama_announcements.create');
+    }
+    
+        /**
+     * Store a newly created resource in storage.
+     *
+     * @return Response
+     */
+    public function store()
+    {
+        // validate
+        // read more on validation at http://laravel.com/docs/validation
+        $rules = array(
+            'title'       => 'required',
+            'text'      => 'required',
+            'user_id',
+            
+        );
+        $validator = Validator::make(Input::all(), $rules);
+
+        // process the login
+        if ($validator->fails()) {
+            return Redirect::to('ama_announcements/create')
+                ->withErrors($validator);
+        } else {
+            // store
+            $ama_announcements = new AmaAnnouncement;
+            $ama_announcements->title       = Input::get('title');
+            $ama_announcements->text      = Input::get('text');
+            $ama_announcements->user_id =  Auth::id();
+            
+            $ama_announcements->save();
+
+            // redirect
+            Session::flash('message', 'Successfully created announcement!');
+            return Redirect::to('ama_announcements');
```

Comment:
You can just do `redirect('ama_announcements')->with('message', 'Successfully created announcement!')` so you don't need to include the Redirect and Session classes. It will make the code cleaner and easier to read. Check [this](https://laravel.com/docs/5.7/redirects#redirecting-with-flashed-session-data) for reference.